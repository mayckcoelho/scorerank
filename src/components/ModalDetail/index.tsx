import React from 'react';
import ModalDefault from "react-native-modal";

import {
    Content,
    Indicator,
    CloseButton,
    CloseText
} from './styles';

interface Props {
    isVisible: boolean;
    close(): void;
}

const ModalDetail: React.FC<Props> = ({ isVisible, children, close }) => {
    return (
        <ModalDefault style={{ margin: 0 }} isVisible={isVisible}>
            <Content>
                <Indicator />
                {children}
                <CloseButton onPress={close}>
                    <CloseText>Fechar</CloseText>
                </CloseButton>
            </Content>
        </ModalDefault>
    )
}

export default ModalDetail;