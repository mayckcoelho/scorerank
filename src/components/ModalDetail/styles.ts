import styled from "styled-components/native";

export const Content = styled.View`
    background-color: #fff;
    position: absolute;
    bottom: 0;
    height: 50%;
    width: 100%;
    margin: 0;
    border-top-left-radius: 15px;
    border-top-right-radius: 15px;
    padding: 0 15px 15px;
    justify-content: space-between;
`

export const Indicator = styled.View`
    width: 50px;
    height: 5px;
    background-color: #ccc;
    border-radius: 50px;
    align-self: center;
    margin-top: 5px;
`

export const CloseButton = styled.TouchableOpacity`
    width: 100%;
    height: 50px;
    border-radius: 10px;
    background-color: #ccc;
    justify-content: center;
    align-items: center;
`

export const CloseText = styled.Text`
    color: #000;
`