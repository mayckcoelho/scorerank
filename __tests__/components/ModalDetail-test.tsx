import React from 'react';
import ModalDetail from '../../src/components/ModalDetail';
import renderer from 'react-test-renderer';

describe('ModalDetail should render correctly', () => {
    it('match snapshot', () => {
        const modal = renderer
            .create(<ModalDetail
                isVisible={true}
                close={() => { }} />)
            .toJSON();
        expect(modal).toMatchSnapshot();
    });
})