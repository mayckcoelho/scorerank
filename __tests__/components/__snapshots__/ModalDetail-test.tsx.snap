// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDetail should render correctly match snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderTopLeftRadius": 15,
            "borderTopRightRadius": 15,
            "bottom": 0,
            "height": "50%",
            "justifyContent": "space-between",
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 0,
            "position": "absolute",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "backgroundColor": "#ccc",
              "borderRadius": 50,
              "height": 5,
              "marginTop": 5,
              "width": 50,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ccc",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Fechar
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
